src/datreant/core/tests/test_trees.py

```diff
@@ -221,6 +221,36 @@ def test_glob(self, tree):
         assert tl in tree.glob('*r*y')
         assert tc in tree.glob('*r*y')
 
+    def test_walk(self, tree, tmpdir):
+        with tmpdir.as_cwd():
+            tree['scipy'].make()
+            tree['2016'].make()
+            tree['sprint'].make()
```

Comment:
Space things out a bit, and please add comments for what some of these blocks are